homeassistant/components/statistics/sensor.py

```diff
@@ -84,7 +91,7 @@
     {
         vol.Required(CONF_ENTITY_ID): cv.entity_id,
         vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
-        vol.Optional(CONF_STATE_CHARACTERISTIC, default=STAT_MEAN): vol.In(
+        vol.Required(CONF_STATE_CHARACTERISTIC): vol.In(
```

Comment:
I'm not sure this is a wanted change, it will force all users to update their configurations.

Reply:
Do users who are using the statistics sensor to average a sensor value have to update their configuration?

According to the breaking change section in https://github.com/home-assistant/core/pull/59867, the breaking change before this PR is that there are no longer any statistic measures exposed as attributes.

Reply:
* **No,** the (default) average/mean functionality of the statistics sensor is not broken by #59867.

* **Yes,** every attribute is removed (hence broken) by #59867. Therefore, every configuration that relates to an attribute will be broken and I *suspect* many of these configurations will be broken without any or a direct warning (thinking of attribute triggers for example).

* **Yes,** the (default) count functionality for `binary_sensor` source sensors would be broken by the PR here.

As I mentioned before, I see this as the last bigger breaking change in the evolution of the statistics sensor. I would certainly be able to avoid this change. I see options to keep the old behavior or to deprecate defaults over time. Overall my main argument is that now would be the best time to rip the band-aid off. 

I understand your hesitance to accept a breaking change. Hope I made my point, maybe you want to discuss internally, please advise.

Reply:
> deprecate defaults over time

That's better, let's go that route.

Reply:
Done.